src/app/[room]/page.tsx

```diff
@@ -1,31 +1,64 @@
 'use client'
 
-import { useEffect, useState } from 'react'
-import SocketIoInfo from '@/components/socketIoDevTools/SocketIoInfo'
+import { useEffect, useState, useRef } from 'react'
 import { socketRoomEmitter } from '@/services/socket'
 import RoomMainUi from '@/components/RoomMainUi'
-import { getRandomName } from '@/utils/sampleData'
+import RoomInfo from '@/components/RoomInfo'
+import NewUserDialog from '@/components/NewUserDialog'
 
-export default function UserRooms({ params }: { params: { room: string } }) {
+type Params = {
+	room: string
+}
+
+export default function UserRooms({ params }: { params: Params }) {
 	const [user, setUser] = useState('')
+	const [displayErrorMessage, setDisplayErrorMessage] = useState(false)
+	const dialogRef = useRef<HTMLDialogElement>(null)
+	console.log('%c>>> params', 'color: red', params)
+
 	const { room } = params
 
 	useEffect(() => {
-		const user = getRandomName()
-		console.log('%c>>> roomId User params:', 'color: red', room, user)
-		setUser(user)
-		socketRoomEmitter('join-room', 'joined room', room, user)
-	}, [room])
+		if (dialogRef.current) {
+			dialogRef.current.showModal()
+		}
+	}, [])
+
+	function handleOnSubmit(formValues: { userName: string }) {
+		console.log('%c>>> formValues', 'color: #5f0', formValues)
+		if (!formValues.userName) {
+			setDisplayErrorMessage(true)
+			return
+		}
+		setUser(formValues.userName)
+		// const timeStamp = Date.now().toString()
```

Comment:
nit Why did you leave this code comment? It can be gone from everywhere :)

Reply:
Already done. It'll be in the next commit.